src/cmd/linuxkit/run_vbox.go

```diff
@@ -0,0 +1,266 @@
+	if err != nil {
+		log.Fatalf("modifyvm --firmware error: %v\n%s", err, out)
+	}
+
+	// set up serial console
+	_, out, err = manage(vboxmanage, "modifyvm", name, "--uart1", "0x3F8", "4")
+	if err != nil {
+		log.Fatalf("modifyvm --uart error: %v\n%s", err, out)
+	}
+
+	var consolePath string
+	if runtime.GOOS == "windows" {
+		// TODO this probably needs to be configurable
+		consolePath = `\\.\pipe\` + name
+	} else {
+		consolePath = filepath.Join(*state, "console")
+		consolePath, err = filepath.Abs(consolePath)
+		if err != nil {
+			log.Fatalf("Bad path: %v", err)
+		}
+	}
+
+	_, out, err = manage(vboxmanage, "modifyvm", name, "--uartmode1", "client", consolePath)
+	if err != nil {
+		log.Fatalf("modifyvm --uartmode error: %v\n%s", err, out)
+	}
+
+	_, out, err = manage(vboxmanage, "storagectl", name, "--name", "IDE Controller", "--add", "ide")
+	if err != nil {
+		log.Fatalf("storagectl error: %v\n%s", err, out)
+	}
+
+	if *isoBoot {
+		_, out, err = manage(vboxmanage, "storageattach", name, "--storagectl", "IDE Controller", "--port", "1", "--device", "0", "--type", "dvddrive", "--medium", path)
+		if err != nil {
+			log.Fatalf("storageattach error: %v\n%s", err, out)
+		}
+		_, out, err = manage(vboxmanage, "modifyvm", name, "--boot1", "dvd")
+		if err != nil {
+			log.Fatalf("modifyvm --boot error: %v\n%s", err, out)
+		}
+	} else {
+		_, out, err = manage(vboxmanage, "storageattach", name, "--storagectl", "IDE Controller", "--port", "1", "--device", "0", "--type", "hdd", "--medium", path)
+		if err != nil {
+			log.Fatalf("storageattach error: %v\n%s", err, out)
+		}
+		_, out, err = manage(vboxmanage, "modifyvm", name, "--boot1", "disk")
+		if err != nil {
+			log.Fatalf("modifyvm --boot error: %v\n%s", err, out)
+		}
+	}
+
+	for i, d := range disks {
+		id := strconv.Itoa(i)
+		if d.Size != 0 && d.Format == "" {
+			d.Format = "raw"
+		}
+		if d.Format != "raw" && d.Path == "" {
+			log.Fatal("vbox currently can only create raw disks")
+		}
+		if d.Path == "" && d.Size == 0 {
+			log.Fatal("please specify an existing disk file or a size")
+		}
+		if d.Path == "" {
+			d.Path = filepath.Join(*state, "disk"+id+".img")
+			if err := os.Truncate(d.Path, int64(d.Size)*int64(1048576)); err != nil {
+				log.Fatalf("Cannot create disk: %v", err)
+			}
+		}
+		_, out, err = manage(vboxmanage, "storageattach", name, "--storagectl", "IDE Controller", "--port", "2", "--device", id, "--type", "hdd", "--medium", d.Path)
+		if err != nil {
+			log.Fatalf("storageattach error: %v\n%s", err, out)
+		}
+	}
+
+	_, out, err = manage(vboxmanage, "modifyvm", name, "--nictype1", "virtio")
+	if err != nil {
+		log.Fatalf("modifyvm --nictype error: %v\n%s", err, out)
+	}
+
+	_, out, err = manage(vboxmanage, "modifyvm", name, "--nic1", *networking)
+	if err != nil {
+		log.Fatalf("modifyvm --nic error: %v\n%s", err, out)
+	}
+
+	// create socket
+	_ = os.Remove(consolePath)
+	ln, err := net.Listen("unix", consolePath)
```

Comment:
This won't work on Windows. Looks like you create a named pipe about and this Listen will fail

Reply:
ah yes